docker-env in github actions: X Error detecting shell: Error: Unknown shell

when minikube is not sure it should try to eval for bash. instead of crashing.
we could warn the user that we are Guessing it is bash.
but in case of github action we know it is bash but minikube can not detect it

```
2020-04-24T05:09:12.2555516Z ##[section]Starting: Request a runner to run this job
2020-04-24T05:09:12.9281804Z Requesting a hosted runner in current repository's account/organization with labels: 'ubuntu-latest', require runner match: True
2020-04-24T05:09:13.1747464Z Labels matched hosted runners has been found, waiting for one of them get assigned for this job.
2020-04-24T05:09:13.2184756Z ##[section]Finishing: Request a runner to run this job
2020-04-24T05:09:18.8935269Z Current runner version: '2.169.1'
2020-04-24T05:09:18.8951499Z ##[group]Operating System
2020-04-24T05:09:18.8951943Z Ubuntu
2020-04-24T05:09:18.8952089Z 18.04.4
2020-04-24T05:09:18.8952226Z LTS
2020-04-24T05:09:18.8952326Z ##[endgroup]
2020-04-24T05:09:18.8952473Z ##[group]Virtual Environment
2020-04-24T05:09:18.8952644Z Environment: ubuntu-18.04
2020-04-24T05:09:18.8952786Z Version: 20200406.2
2020-04-24T05:09:18.8952952Z Included Software: https://github.com/actions/virtual-environments/blob/ubuntu18/20200406.2/images/linux/Ubuntu1804-README.md
2020-04-24T05:09:18.8953132Z ##[endgroup]
2020-04-24T05:09:18.8953818Z Prepare workflow directory
2020-04-24T05:09:18.9107300Z Prepare all required actions
2020-04-24T05:09:18.9115080Z Download action repository 'medyagh/minikube-action@master'
2020-04-24T05:09:19.7571501Z ##[group]Run medyagh/minikube-action@master
2020-04-24T05:09:19.7571808Z with:
2020-04-24T05:09:19.7572058Z   action: start
2020-04-24T05:09:19.7572204Z   driver: docker
2020-04-24T05:09:19.7572372Z ##[endgroup]
2020-04-24T05:09:20.6871637Z Downloading Minikube from https://github.com/kubernetes/minikube/releases/download/v1.10.0-beta.1/minikube-linux-amd64
2020-04-24T05:09:20.8941518Z [command]/bin/chmod +x /home/runner/work/_temp/ccd5d601-f9cb-4ce7-8ff5-8b18ff0a001f
2020-04-24T05:09:20.9066703Z Starting minikube
2020-04-24T05:09:20.9069152Z [command]/home/runner/bin/minikube start
2020-04-24T05:09:20.9493535Z * minikube v1.10.0-beta.1 on Ubuntu 18.04
2020-04-24T05:09:21.4809757Z * Automatically selected the docker driver
2020-04-24T05:09:21.5691732Z * Starting control plane node minikube in cluster minikube
2020-04-24T05:09:22.1830977Z * Pulling base image ...
2020-04-24T05:09:22.1832937Z * Downloading Kubernetes v1.18.0 preload ...
2020-04-24T05:09:50.3831434Z     > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 13.73 MiB     > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 46.88 MiB     > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 76.81 MiB     > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 96.00 MiB     > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 121.22 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 140.75 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 176.12 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 193.22 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 216.89 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 233.39 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 254.47 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 278.00 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 309.34 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 328.00 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 346.56 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 374.88 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 392.00 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 414.73 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 451.16 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 467.00 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 503.28 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 523.66 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 541.22 MiB    > preloaded-images-k8s-v2-v1.18.0-docker-overlay2-amd64.tar.lz4: 542.91 MiB* Creating docker container (CPUs=2, Memory=2200MB) ...
2020-04-24T05:10:17.2600171Z * Preparing Kubernetes v1.18.0 on Docker 19.03.2 ...
2020-04-24T05:10:17.2604240Z   - kubeadm.pod-network-cidr=10.244.0.0/16
2020-04-24T05:10:30.6753595Z * Verifying Kubernetes components...
2020-04-24T05:10:31.2359457Z * Enabled addons: default-storageclass, storage-provisioner
2020-04-24T05:10:31.2393115Z * Done! kubectl is now configured to use "minikube"
2020-04-24T05:10:36.9344735Z minikube started successfully.
2020-04-24T05:10:36.9359697Z Finished starting minikube
2020-04-24T05:10:36.9434853Z ##[group]Run kubectl get pods -A
2020-04-24T05:10:36.9435045Z [36;1mkubectl get pods -A[0m
2020-04-24T05:10:36.9468653Z shell: /bin/bash -e {0}
2020-04-24T05:10:36.9468753Z ##[endgroup]
2020-04-24T05:10:37.5489163Z NAMESPACE     NAME                               READY   STATUS    RESTARTS   AGE
2020-04-24T05:10:37.5490384Z kube-system   etcd-minikube                      0/1     Pending   0          7s
2020-04-24T05:10:37.5525456Z kube-system   kube-apiserver-minikube            0/1     Pending   0          7s
2020-04-24T05:10:37.5526355Z kube-system   kube-controller-manager-minikube   1/1     Running   0          7s
2020-04-24T05:10:37.5527325Z kube-system   kube-scheduler-minikube            1/1     Running   0          6s
2020-04-24T05:10:37.5530069Z kube-system   storage-provisioner                0/1     Pending   0          6s
2020-04-24T05:10:37.5607380Z ##[group]Run docker ps
2020-04-24T05:10:37.5607595Z [36;1mdocker ps[0m
2020-04-24T05:10:37.5607702Z [36;1meval $(minikube -p minikube docker-env)[0m
2020-04-24T05:10:37.5607809Z [36;1mdocker ps[0m
2020-04-24T05:10:37.5642021Z shell: /bin/bash -e {0}
2020-04-24T05:10:37.5642109Z ##[endgroup]
2020-04-24T05:10:38.2029674Z * 
2020-04-24T05:10:38.2029934Z CONTAINER ID        IMAGE                                COMMAND                  CREATED             STATUS              PORTS                                                                                                      NAMES
2020-04-24T05:10:38.2047820Z X Error detecting shell: Error: Unknown shell
2020-04-24T05:10:38.2048885Z a440fc123610        gcr.io/k8s-minikube/kicbase:v0.0.9   "/usr/local/bin/entr…"   45 seconds ago      Up 41 seconds       127.0.0.1:32771->22/tcp, 127.0.0.1:32770->2376/tcp, 127.0.0.1:32769->5000/tcp, 127.0.0.1:32768->8443/tcp   minikube
2020-04-24T05:10:38.2049276Z * 
2020-04-24T05:10:38.2049578Z * minikube is exiting due to an error. If the above message is not useful, open an issue:
2020-04-24T05:10:38.2050028Z   - https://github.com/kubernetes/minikube/issues/new/choose
2020-04-24T05:10:38.2050501Z /home/runner/work/_temp/0e91234d-00da-4304-92f8-2fc89a013607.sh: eval: line 2: unexpected EOF while looking for matching `''
2020-04-24T05:10:38.2064104Z ##[error]Process completed with exit code 2.
2020-04-24T05:10:38.2075933Z Cleaning up orphan processes
```

I event set the shell: bash explicitly and still no luck

I even set shell: bash expclicitly but still minikube cant detect it
https://github.com/k8sguard/discover/blob/master/.github/workflows/main.yml#L17

--- a/pkg/minikube/shell/shell.go
+++ b/pkg/minikube/shell/shell.go
@@ -21,6 +21,8 @@ package shell
 import (
 	"fmt"
 	"io"
+	"os"
+	"runtime"
 	"strings"
 	"text/template"
 
@@ -139,13 +141,12 @@ var (
 
 // Detect detects user's current shell.
 func Detect() (string, error) {
-	sh, err := shell.Detect()
+	sh := os.Getenv("SHELL")
 	// Don't error out when $SHELL has not been set
-	// Note: ErrUnknownShell is not used on windows
-	if sh == "" {
+	if sh == "" && runtime.GOOS != "windows" {
 		return defaultSh, nil
 	}
-	return sh, err
+	return shell.Detect()
 }
 
 func (c EnvConfig) getShell() shellData {

--- a/pkg/minikube/shell/shell.go
+++ b/pkg/minikube/shell/shell.go
@@ -1,6 +1,5 @@
 /*
 Copyright 2020 The Kubernetes Authors All rights reserved.
-
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
@@ -130,7 +129,8 @@ REM @FOR /f "tokens=*" %%i IN ('%s') DO @%%i
 	},
 }
 
-var defaultShell shellData = shellConfigMap["bash"]
+var defaultSh = "bash"
+var defaultShell shellData = shellConfigMap[defaultSh]
 
 var (
 	// ForceShell forces a shell name
@@ -139,7 +139,13 @@ var (
 
 // Detect detects user's current shell.
 func Detect() (string, error) {
-	return shell.Detect()
+	sh, err := shell.Detect()
+	// Don't error out when $SHELL has not been set
+	// Note: ErrUnknownShell is not used on windows
+	if sh == "" {
+		return defaultSh, nil
+	}
+	return sh, err
 }
 
 func (c EnvConfig) getShell() shellData {

--- a/pkg/minikube/shell/shell.go
+++ b/pkg/minikube/shell/shell.go
@@ -1,5 +1,6 @@
 /*
 Copyright 2020 The Kubernetes Authors All rights reserved.
+
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

--- a/pkg/minikube/shell/shell_test.go
+++ b/pkg/minikube/shell/shell_test.go
@@ -117,11 +117,23 @@ set -e bar;`},
 	}
 }
 
-func TestDetect(t *testing.T) {
+func TestDetectSet(t *testing.T) {
 	orgShellEnv := os.Getenv("SHELL")
 	defer os.Setenv("SHELL", orgShellEnv)
 
-	os.Setenv("SHELL", "bash")
+	os.Setenv("SHELL", "/bin/bash")
+	if s, err := Detect(); err != nil {
+		t.Fatalf("unexpected error: '%v' during shell detection. Returned shell: %s", err, s)
+	} else if s == "" {
+		t.Fatalf("Detected shell expected to be non empty string")
+	}
+}
+
+func TestDetectUnset(t *testing.T) {
+	orgShellEnv := os.Getenv("SHELL")
+	defer os.Setenv("SHELL", orgShellEnv)
+
+	os.Unsetenv("SHELL")
 	if s, err := Detect(); err != nil {
 		t.Fatalf("unexpected error: '%v' during shell detection. Returned shell: %s", err, s)
 	} else if s == "" {

